File: exercism/python/word-count/word_count.py
# https://exercism.io/my/solutions/47314adde6f04a67a6fd3f8082d61d52

from collections import Counter
from re import sub


def count_words(sentence):
    words = sub('[^0-9a-z\']+', ' ', sentence.lower()).split()
    no_quotes = map(lambda w: w.strip("'"), words)
    return Counter(no_quotes)

File: exercism/go/word-count/word_count.go
// https://exercism.io/my/solutions/e5e65127ef9f4458829ce0955ddc2be4

package wordcount

import (
	"regexp"
	"strings"
)

// Frequency represents a counter dictionary
type Frequency map[string]int

// Clean removes non-alphanumerical characters from a string
func clean(s string) string {
	re, _ := regexp.Compile("[^a-zA-Z0-9'\t\n]+")
	return re.ReplaceAllLiteralString(s, " ")
}

// WordCount counts the words in a sentence
func WordCount(sentence string) Frequency {
	counter := Frequency{}
	for _, w := range strings.Fields(clean(sentence)) {
		word := strings.ToLower(strings.Trim(w, "'\""))
		counter[word]++
	}
	return counter
}
